rfcs/proposed/task_aggregation/README.md

```diff
@@ -0,0 +1,400 @@
+# Task Aggregation
+
+This proposal outlines a possible solution to producer-consumer inefficiencies in the current ``task_group`` API.
+
+## Table of Contents
+
+* 1 [Introduction](#introduction)
+* 2 [Proposed Design](#proposed-design)
+  * 2.1 [Resolving the Single Task Source Bottleneck](#resolving-the-single-task-source-bottleneck)
+  * 2.2 [Resolving the Reference Counting Bottleneck](#resolving-the-reference-counting-bottleneck)
+  * 2.3 [Allocator Bottleneck](#allocator-bottleneck)
+  * 2.4 [Grabbing Thread Slowdown](#grabbing-thread-slowdown)
+* 3 [Performance Analysis](#performance-analysis)
+* 4 [Possible APIs](#possible-apis)
+* 5 [Open Questions](#open-questions)
+* 6 [Task Tree Operations in Detail](#task-tree-operations-in-detail)
+  * 6.1 [Synchronization Between Grabber and Producer](#synchronization-between-grabber-and-producer)
+  * 6.2 [Adding a Task to the Tree](#adding-a-task-to-the-tree)
+  * 6.3 [Splitting the Tree](#splitting-the-tree)
+* 7 [Alternative Approaches Considered](#alternative-approaches-considered)
+
+## Introduction
+
+Consider an application that uses ``tbb::task_group`` with a single-producer/multiple-consumers pattern: one producer thread generates work items, and 
+multiple consumer threads process them in parallel.
+
+```cpp
+
+void produce() {
+    tbb::task_group tg;
+
+    auto work_item = get_work_item();
+
+    while (work_item != end_of_work) {
+        tg.run(process(work_item)); // Can be executed by any worker
```

Comment:
Is it necessary that the work are calls to the same function, i.e. "process(work_item)" or can the tasks be different?